crates/wdl-engine/src/eval/v1.rs

```diff
@@ -31,3 +40,61 @@ fn write_json_file(path: impl AsRef<Path>, value: &impl Serialize) -> Result<()>
     serde_json::to_writer_pretty(BufWriter::new(file), value)
         .with_context(|| format!("failed to write file `{path}`", path = path.display()))
 }
+
+/// The top-level evaluation context.
+///
+/// "Top-level" here means the outermost invocation of a task or workflow across
+/// an entire execution. This type is suitable for once-per-execution values
+/// like a shared container image cache, and new instances should not be created
+/// for evaluating subgraphs of the initial execution.
+///
+/// This type is meant to be cheaply cloned and sendable between threads. When
+/// adding to it, make sure to use an `Arc` for any non-trivially-sized data.
+#[derive(Clone)]
+pub struct TopLevelEvaluator {
```

Comment:
The thing I like least about all of this is right here: I cannot figure out a good name for this. I think it might have to wait for future refactoring of `sprocket::eval` so we can steal the name `Evaluator`, but everything else I could think of was already taken. Suggestions are very welcome.

Reply:
Or what about just `Evaluator`?

Reply:
Hm. I guess, given all of this, I would prefer to just stay with `TopLevelEvaluator` and reconsider this question more fully when we do the interface refactor of `wdl-engine`. I don't think this question should block this PR and the release we hope to do soon.